src/parser/mod.rs

```diff
@@ -6193,24 +6186,160 @@ impl<'a> Parser<'a> {
         {
             self.parse_optional_column_option_as()
+        } else if self.parse_keyword(Keyword::AUTOINCREMENT)
+            && dialect_of!(self is SnowflakeDialect | SQLiteDialect |  GenericDialect)
+        {
+            if dialect_of!(self is SnowflakeDialect) {
+                self.prev_token();
+                return self.parse_snowflake_autoincrement_or_identity_option_column();
+            }
+
+            // Support AUTOINCREMENT for SQLite
+            Ok(Some(ColumnOption::DialectSpecific(vec![
+                Token::make_keyword("AUTOINCREMENT"),
+            ])))
         } else if self.parse_keyword(Keyword::IDENTITY)
-            && dialect_of!(self is MsSqlDialect | GenericDialect)
+            && dialect_of!(self is MsSqlDialect | SnowflakeDialect | GenericDialect)
         {
-            let property = if self.consume_token(&Token::LParen) {
+            if dialect_of!(self is SnowflakeDialect) {
+                self.prev_token();
+                return self.parse_snowflake_autoincrement_or_identity_option_column();
+            }
+
+            let parameters = if self.consume_token(&Token::LParen) {
                 let seed = self.parse_number()?;
                 self.expect_token(&Token::Comma)?;
                 let increment = self.parse_number()?;
                 self.expect_token(&Token::RParen)?;
 
-                Some(IdentityProperty { seed, increment })
+                Some(IdentityFormat::FunctionCall(IdentityParameters {
+                    seed,
+                    increment,
+                }))
             } else {
                 None
             };
-            Ok(Some(ColumnOption::Identity(property)))
+            Ok(Some(ColumnOption::Identity(Identity::Identity(
+                IdentityProperty {
+                    parameters,
+                    order: None,
+                },
+            ))))
+        } else if ((self.parse_keyword(Keyword::WITH)
+            && self
+                .parse_one_of_keywords(&[Keyword::MASKING, Keyword::PROJECTION])
+                .is_some())
+            || self
+                .parse_one_of_keywords(&[Keyword::MASKING, Keyword::PROJECTION])
+                .is_some())
+            && dialect_of!(self is SnowflakeDialect | GenericDialect)
+        {
+            self.prev_token();
+            let Some(policy) = self.parse_snowflake_column_policy()? else {
+                return Ok(None);
+            };
+            Ok(Some(ColumnOption::Policy(policy)))
+        } else if self.parse_keywords(&[Keyword::TAG])
+            && dialect_of!(self is SnowflakeDialect | GenericDialect)
+        {
+            self.expect_token(&Token::LParen)?;
+            let tags = self.parse_comma_separated(Self::parse_tag)?;
+            self.expect_token(&Token::RParen)?;
+
+            Ok(Some(ColumnOption::Tags(tags)))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn parse_snowflake_autoincrement_or_identity_option_column(
```

Comment:
A similar comment on these ones, thinking it'll help with a comment showing the syntax/example of what's being parsed, ideally with a link to the doc?

Reply:
I added a reasonable documentation for structs with links to a Snowflake doc and moved this function into `snowflake` dialect implementation.